aws/resource_aws_ses_identity_notification_test.go

```diff
@@ -0,0 +1,123 @@
+	domain := fmt.Sprintf(
+		"%s.terraformtesting.com",
+		acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsSESIdentityNotificationDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: fmt.Sprintf(testAccAwsSESIdentityNotificationConfig_basic, domain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsSESIdentityNotificationExists("aws_ses_identity_notification.test"),
+				),
+			},
+			resource.TestStep{
+				Config: fmt.Sprintf(testAccAwsSESIdentityNotificationConfig_update, domain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsSESIdentityNotificationExists("aws_ses_identity_notification.test"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsSESIdentityNotificationDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).sesConn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_ses_identity_notification" {
+			continue
+		}
+
+		identity := rs.Primary.ID
+		params := &ses.GetIdentityNotificationAttributesInput{
+			Identities: []*string{aws.String(identity)},
+		}
+
+		response, err := conn.GetIdentityNotificationAttributes(params)
+		if err != nil {
+			return err
+		}
+
+		if response.NotificationAttributes[identity] != nil {
+			return fmt.Errorf("SES Identity Notification %s still exists. Failing!", identity)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAwsSESIdentityNotificationExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("SES Identity Notification not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("SES Identity Notification identity not set")
+		}
+
+		identity := rs.Primary.ID
+		conn := testAccProvider.Meta().(*AWSClient).sesConn
+
+		params := &ses.GetIdentityNotificationAttributesInput{
+			Identities: []*string{aws.String(identity)},
+		}
+
+		response, err := conn.GetIdentityNotificationAttributes(params)
+		if err != nil {
+			return err
+		}
+
+		if response.NotificationAttributes[identity] == nil {
+			return fmt.Errorf("SES Identity Notification %s not found in AWS", identity)
+		}
+
+		return nil
+	}
+}
+
+const testAccAwsSESIdentityNotificationConfig_basic = `
+resource "aws_ses_identity_notification" "test" {
+	identity = "${aws_ses_identity.test.arn}"
+	notification_type = "Complaint"
+}
+
+resource "aws_ses_identity" "test" {
+  domain = "%s"
+}
+`
+const testAccAwsSESIdentityNotificationConfig_update = `
+resource "aws_ses_identity_notification" "test" {
+	topic_arn = "${aws_sns_topic.test.arn}"
+	identity = "${aws_ses_identity.test.arn}"
+	notification_type = "Complaint"
+}
+
+resource "aws_ses_identity" "test" {
+  domain = "%s"
+}
+
+resource "aws_sns_topic" "test" {
+  name = "user-updates-topic"
```

Comment:
This topic name needs to be randomized for parallelization or if the test fails and leaves the topic dangling: `name = "%s"`

In the acceptance test function, you can use something like `rName := fmt.Sprintf("tf-acc-test-%d", acctest.RandInt())` to get a random topic name.